test/unittest/xmlparser/XMLEndpointParserTests.cpp

```diff
@@ -0,0 +1,862 @@
+                "\
+                <writer>\
+                    <userId>3</userId>\
+                    <entityID>4</entityID>\
+                    <expectsInlineQos>true</expectsInlineQos>\
+                    <topicName>HelloWorldTopic</topicName>\
+                    <topicDataType>HelloWorld</topicDataType>\
+                    <topicKind>NO_KEY</topicKind>\
+                    <topic name=\"HelloWorldTopic\" dataType=\"HelloWorld\" kind=\"NO_KEY\"/>\
+                    <partitionQos>HelloPartition</partitionQos>\
+                    <partitionQos>WorldPartition</partitionQos>\
+                    <unicastLocator address=\"192.168.0.128\" port=\"5000\"/>\
+                    <unicastLocator address=\"10.47.8.30\" port=\"6000\"/>\
+                    <multicastLocator address=\"239.255.1.1\" port=\"7000\"/>\
+                    <reliabilityQos>BEST_EFFORT_RELIABILITY_QOS</reliabilityQos>\
+                    <durabilityQos>VOLATILE_DURABILITY_QOS</durabilityQos>\
+                    <ownershipQos kind=\"SHARED_OWNERSHIP_QOS\"/>\
+                    <livelinessQos kind=\"AUTOMATIC_LIVELINESS_QOS\" leaseDuration_ms=\"1000\"/>\
+                </writer>\
+                ";
+        ASSERT_EQ(tinyxml2::XMLError::XML_SUCCESS, xml_doc.Parse(xml));
+        titleElement = xml_doc.FirstChildElement();
+        EXPECT_EQ(XMLP_ret::XML_OK, mp_edpXML->loadXMLWriterEndpoint(titleElement, pdata));
+
+        // Topic attributes
+        EXPECT_EQ(pdata->m_writers[0]->topicName(), "HelloWorldTopic");
+        EXPECT_EQ(pdata->m_writers[0]->topicKind(), TopicKind_t::NO_KEY);
+        EXPECT_EQ(pdata->m_writers[0]->typeName(), "HelloWorld");
+        EXPECT_EQ(pdata->m_writers[0]->has_locators(), true);
+
+        // Locators
+        Locator_t uni_loc;
+        IPLocator::setIPv4(uni_loc, "192.168.0.128");
+        uni_loc.port = static_cast<uint16_t>(5000);
+        EXPECT_EQ(pdata->m_writers[0]->remote_locators().unicast[0],  uni_loc);
+
+        Locator_t multi_loc;
+        IPLocator::setIPv4(multi_loc, "239.255.1.1");
+        multi_loc.port = static_cast<uint16_t>(7000);
+        EXPECT_EQ(pdata->m_writers[0]->remote_locators().multicast[0],  multi_loc);
+
+        // qos
+        EXPECT_EQ(pdata->m_writers[0]->m_qos.m_reliability.kind,  BEST_EFFORT_RELIABILITY_QOS);
+        EXPECT_EQ(pdata->m_writers[0]->m_qos.m_durability.kind,  VOLATILE_DURABILITY_QOS);
+        EXPECT_EQ(pdata->m_writers[0]->m_qos.m_ownership.kind,  SHARED_OWNERSHIP_QOS);
+        EXPECT_EQ(pdata->m_writers[0]->m_qos.m_liveliness.kind,  AUTOMATIC_LIVELINESS_QOS);
+
+        // Delete the WriterProxyData created inside loadXMLWriterEndpoint
+        delete pdata->m_writers[0];
+
+        // Then delete StaticRTPSParticipantInfo
+        delete pdata;
+    }
+
+    {
+        StaticRTPSParticipantInfo* pdata = new StaticRTPSParticipantInfo();
+        // Parametrized XML
+        const char* xml_p =
+                "\
+                <reader>\
```

Comment:
```suggestion
                <writer>\
```
Fix this error along `loadXMLWriterEndpoint` function